maintenance/wikia/email.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * Email
+ *
+ * This sends an email type (defined in extensions/wikia/Email) to the email address of your
+ * choice
+ *
+ */
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_NOTICE );
+
+// Eliminate the need to set this on the command line
+if ( !getenv( 'SERVER_ID' ) ) {
+	putenv( "SERVER_ID=177" );
+}
+
+require_once( dirname( __FILE__ ) . '/../Maintenance.php' );
+
+/**
+ * Class Email\CLI
+ */
+class EmailCLI extends Maintenance {
+
+	protected $verbose = false;
+	protected $test = false;
+	protected $to = '';
+	protected $type = '';
+	protected $argString = '';
+	protected $list = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Pre-populate LVS suggestions";
+		$this->addOption( 'test', 'Test mode; make no changes',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG,
+			't'
+		);
+		$this->addOption( 'verbose', 'Show extra debugging output',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG,
+			'v'
+		);
+
+		$this->addOption( 'to', 'Address to send email to',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'type', 'Type of email to send',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'args', 'Arguments for email type',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_HAS_ARG
+		);
+		$this->addOption( 'list', 'List email types',
+			Maintenance::PARAM_OPTIONAL,
+			Maintenance::PARAM_IS_FLAG
+		);
+	}
+
+	public function execute() {
+		try {
+			$this->initOptions();
+
+			if ( $this->list ) {
+				$this->showValidEmailTypes();
+				exit( 0 );
+			}
+
+			// The controller class should end with 'Controller'
+			$controllerClass = 'Email\\Controller\\' . $this->type . 'Controller';
+			$this->assertControllerExists( $controllerClass );
+
+			$params = $this->argStringToArray( $this->argString );
+			$resp = F::app()->sendRequest( $controllerClass, 'handle', $params, true );
+
+			$this->assertSuccessfulEmail( $resp );
+
+			echo "Email sent as '$this->to' successfully!\n";
+			exit( 0 );
+		} catch ( Exception $e ) {
```

Comment:
I know it's command line tool, but when I see statement like this I think only about one thing:
Pokémon Exception Handling
http://blog.codinghorror.com/new-programming-jargon/

This is big block of code and many unexpected exceptions can be thrown. Do you think it's best idea to limit output only to error message? Personally when running CLI script and something unexpected goes wrong I would like to know where exactly problem lays, but maybe I'm strange ;)